pkg/rbd/nodeserver.go

```diff
@@ -146,6 +147,26 @@ func (ns *NodeServer) NodeStageVolume(ctx context.Context, req *csi.NodeStageVol
 	}
 	klog.V(4).Infof(util.Log(ctx, "rbd image: %s/%s was successfully mapped at %s\n"), req.GetVolumeId(), volOptions.Pool, devicePath)
 
+	if volOptions.Encrypted {
```

Comment:
could you recommend where this documentation should be added? from what I can see existing deploy-rbd.md only goes through simple case without any branching or specifics of some options. Should I add a new document for this?

Reply:
Note: `volume_options` in the `NodeStageVolume` request is what is returned as the volume context from CreateVolume request, and it is not directly from the StorageClass (or other such constructs in other COs). Thus the limitation to not change existing StorageClasses to make then dynamically provision encrypted volumes in the future would be incorrect. IOW, all volumes generated prior to adding the `encrypted` parameter to the StorageClass will remain unencrypted and not carry the `encrypted` parameter in their respective `volume_options` and all future volumes would. We can remove riders around the same in the documentation.

Reply:
I do believe this is true, but I think it's better to mention this in documentation still that it's not supported as it will be very confusing for users after changing existing storage class to `encrypted: true` and later seeing some volumes encrypted and some not

or else we could explain the behaviour in the documentation (this won't reduce confusion I think, but at least there will be a place in documentation where we can point users if they have questions around the topic)

Reply:
I agree to the above, I just did not want documentation to sound like, "if this is changed older volumes will no longer work" or some such. To reduce ambiguity it is better never to edit an StorageClass anyway, if possible some rewording in the deploy-rbd.md file can be made.

Bottom line I wanted ensure we are in sync on why a new StorageClass is needed, and what happens if we edit older ones.

Reply:
if you could suggest better wording I could change that, for now I think about "not supported" as "you're on your own if you do this". Explaining each case on how it will work will be documenting too deep into internals and will only add more confusion around

Reply:
Ack! Let's leave the doc as is "not supported" is good enough.